cmd/telliot/cmds.go

```diff
@@ -0,0 +1,456 @@
+		return errors.Wrapf(err, "parsing amount argument")
+	}
+	return ops.Transfer(ctx, logger, client, contract, account, address.addr, amount.Int)
+}
+
+type approveCmd tokenCmd
+
+func (c *approveCmd) Run() error {
+	cfg, err := parseConfig(string(c.Config))
+	if err != nil {
+		return errors.Wrapf(err, "creating config")
+	}
+
+	logger, err := createLogger(cfg.Logger, cfg.LogLevel)
+	if err != nil {
+		return errors.Wrapf(err, "creating logger")
+	}
+
+	ctx := context.Background()
+	client, contract, account, err := createTellorVariables(ctx, cfg)
+	if err != nil {
+		return errors.Wrapf(err, "creating tellor variables")
+	}
+
+	address := ETHAddress{}
+	err = address.Set(c.Address)
+	if err != nil {
+		return errors.Wrapf(err, "parsing address argument")
+	}
+	amount := TRBAmount{}
+	err = amount.Set(c.Amount)
+	if err != nil {
+		return errors.Wrapf(err, "parsing amount argument")
+	}
+	return ops.Approve(ctx, logger, client, contract, account, address.addr, amount.Int)
+}
+
+type balanceCmd struct {
+	Config  configPath `required:"" type:"existingfile" help:"path to config file"`
+	Address string     `arg:"" optional:""`
+}
+
+func (b *balanceCmd) Run() error {
+	cfg, err := parseConfig(string(b.Config))
+	if err != nil {
+		return errors.Wrapf(err, "creating config")
+	}
+
+	_, err = createLogger(cfg.Logger, cfg.LogLevel)
+	if err != nil {
+		return errors.Wrapf(err, "creating logger")
+	}
+
+	ctx := context.Background()
+	client, contract, _, err := createTellorVariables(ctx, cfg)
+	if err != nil {
+		return errors.Wrapf(err, "creating tellor variables")
+	}
+
+	addr := ETHAddress{}
+	if b.Address == "" {
+		err = addr.Set(contract.Address.String())
+		if err != nil {
+			return errors.Wrapf(err, "parsing argument")
+		}
+	} else {
+		err = addr.Set(b.Address)
+		if err != nil {
+			return errors.Wrapf(err, "parsing argument")
+		}
+	}
+	return ops.Balance(ctx, client, contract.Getter, addr.addr)
+}
+
+type stakeCmd struct {
+	Config    configPath `required:"" type:"existingfile" help:"path to config file"`
+	Operation string     `arg:"" required:""`
+}
+
+func (s *stakeCmd) Run() error {
+	cfg, err := parseConfig(string(s.Config))
+	if err != nil {
+		return errors.Wrapf(err, "creating config")
+	}
+
+	logger, err := createLogger(cfg.Logger, cfg.LogLevel)
+	if err != nil {
+		return errors.Wrapf(err, "creating logger")
+	}
+
+	ctx := context.Background()
+	client, contract, account, err := createTellorVariables(ctx, cfg)
+	if err != nil {
+		return errors.Wrapf(err, "creating tellor variables")
+	}
+
+	switch s.Operation {
+	case "deposit":
+		return ops.Deposit(ctx, logger, client, contract, account)
+	case "withdraw":
+		return ops.WithdrawStake(ctx, logger, client, contract, account)
+	case "request":
+		return ops.RequestStakingWithdraw(ctx, logger, client, contract, account)
+	case "status":
+		return ops.ShowStatus(ctx, logger, client, contract, account)
+	default:
+		return errors.New("unknown stake command")
+	}
+}
```

Comment:
I think there should be a way to add subcommand without using string macthing.

Reply:
I think the only other way is to do it like dispute command, and each subcommand gets its own struct and such. Since these have the same requirements, I think make sense to group them together,but let me know if you think it's better the other way. Also, we can add to the help string the possible flags.

Reply:
Yeah these need to be consistent.

command --flags....

It is strange to see that some commands take the flags before the command like how it was for 

telliot --config  mine

Instead, all commands should be in the following format

telliot stake --config
telliot mine --config
telliot deposit --config

As long as this doesn't lead to a lot of duplicated code this makes it easier to remove/refactor/add commands.

Reply:
Oh I see the problem. I will make the commands inconsistent indeed. I'll adjust that quickly